pyccel/parser/semantic.py

```diff
@@ -1030,51 +1032,55 @@ def _extract_indexed_from_var(self, var, args, name):
         # case of Pyccel ast Variable, IndexedVariable
         # if not possible we use symbolic objects
 
+        if not isinstance(var, Variable):
+            assert(hasattr(var,'__getitem__'))
+            if len_args==1:
+                return var[args[0]]
+            else:
+                return self._visit(Indexed(var[args[0]],args[1:]))
+
+        if var.order == 'C':
+            args = args[::-1]
+        args = tuple(args)
+
         if isinstance(var, TupleVariable) and not var.is_homogeneous:
 
-            for i, arg in enumerate(args[::-1]):
-                if (not isinstance(arg, Integer) and
-                        not isinstance(arg, Slice)):
+            arg = args[-1]
+            if (not isinstance(arg, Integer) and
+                    not isinstance(arg, Slice)):
```

Comment:
Can't this be written as follows?
```
if not isinstance(arg, (Integer, Slice)):
    ...
```

Reply:
This is outdated code. Are you looking at a specific commit rather than the final code?

Reply:
Sorry, that must have been the case